### PR TITLE
Refactor the clip embedding tutorial

### PR DESCRIPTION
**Pull request by zhiqiangdon**
*Issue #, if available:*

*Description of changes:*

1. Refactor the CLIP tutorial with the new matcher APIs. Change it a zero shot image-text matching tutorial to align with the CLIP fine-tuning tutorial.
2. Add the keyword `semantic` to titles to be accurate.
3. Fix several inconsistencies between the index page and tutorials of matching.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by bryanyzhu — APPROVED** [review]
LGTM

**Review by cheungdaven — APPROVED** [review]
LGTM!

File: docs/tutorials/multimodal/matching/image_text_matching.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
-# Image<-->Text Matching with AutoMM
+# Image<-->Text Semantic Matching with AutoMM
```
Comment (sxjscience) [inline docs/tutorials/multimodal/matching/image_text_matching.md:1]: We can consider to use `Image-Text Matching with AutoMM`

File: docs/tutorials/multimodal/matching/index.rst [hunk 2]
```diff
@@ -3,41 +3,41 @@ Matching
...
-      :title: Text-to-text Matching with AutoMM - Quick Start
+      :title: Text-to-text Semantic Matching with AutoMM
       :link: text2text_matching.html
 
-      How to use AutoMM for text to text matching.
+      How to use AutoMM for text-to-text semantic matching.
    
    .. card::
-      :title: Image-to-image Matching with AutoMM - Quick Start
+      :title: Image-to-image Semantic Matching with AutoMM
```
Comment (sxjscience) [inline docs/tutorials/multimodal/matching/index.rst:12]: You can do another pass to ensure that the title in the cards are the same as the title of the individual notebooks.

File: docs/tutorials/multimodal/matching/image_text_matching.md (hunk 1 above)
Comment (zhiqiangdon) [reply docs/tutorials/multimodal/matching/image_text_matching.md]: Done.

File: docs/tutorials/multimodal/matching/index.rst (hunk 2 above)
Comment (zhiqiangdon) [reply docs/tutorials/multimodal/matching/index.rst]: Checked.

**Pull request merged**